lib/win32/certstore/store_base.rb

```diff
@@ -125,11 +111,30 @@ def cert_get_name_args(pcert_context, cert_name)
         [pcert_context, CERT_NAME_FRIENDLY_DISPLAY_TYPE, CERT_NAME_ISSUER_FLAG, nil, cert_name, 1024]
       end
 
+      # Remove extra space and : from thumbprint
+      def update_thumbprint(certificate_thumbprint)
+        certificate_thumbprint.gsub(/[^A-Za-z0-9]/, '')
+      end
+
       # Convert OpenSSL::X509::Certificate object in .der formate
       def der_cert(cert_obj)
         FFI::MemoryPointer.from_string(cert_obj.to_der)
       end
 
+      # Get certificate pem
+      def get_cert_pem(thumbprint)
+        IO::popen(["powershell.exe", cert_ps_cmd(thumbprint)]) {|io| io.readlines}
```

Comment:
ideally we do what we need with a ruby library, and then with FFI if there is no library, and finally a shell-out as a last resort. Is there a reason you're not using mixlib-shellout here? We've put a lot of work into shellout to make sure we can capture things like stdin, stdout, error codes, etc. reliably in different circumstances,